### PR TITLE
Do not add members of address to contracts in experimental 0.5.0

### LINKED ISSUE #2683: Remove address overloads on contracts
I think it would make sense removing those overloads and making their use explicit by typecasting to an address.

The reason is to make it explicit what functions are defined on a contract and not have confusing overloads, common examples are in the case of `transfer` and `balance`.

### PR DESCRIPTION
**Pull request by axic**
Fixes #2683. Depends on #2936.

### CODE REVIEW

**Comment by axic** [comment]
Needs tests, but this must be the simplest 0.5.0 change :)

**Comment by axic** [comment]
@chriseth this is ready

File: libsolidity/ast/Types.cpp [hunk 1]
```diff
@@ -1660,7 +1660,9 @@ MemberList::MemberMap ContractType::nativeMembers(ContractDefinition const*) con
 				&it.second->declaration()
 			));
 	}
-	addNonConflictingAddressMembers(members);
+	// In 0.5.0 address members are not populated into the contract.
+	if (!_contract->sourceUnit().annotation().experimentalFeatures.count(ExperimentalFeature::V050))
```
Comment (axic) [inline libsolidity/ast/Types.cpp:1665]: Should we assert for `_contract != nullptr` or turn on 0.5.0 only if it was supplied?

File: docs/types.rst [hunk 2]
```diff
@@ -107,6 +107,9 @@ Operators:
 
 * ``<=``, ``<``, ``==``, ``!=``, ``>=`` and ``>``
 
+.. note::
+    Starting with version 0.5.0 contracts do not derive from the address type.
```
Comment (chriseth) [inline docs/types.rst]: ... but are still convertible to address.

File: libsolidity/ast/Types.cpp (hunk 1 above)
Comment (chriseth) [reply libsolidity/ast/Types.cpp:1665]: `_contract` is different from `m_contract`. The parameter only specifies the current scope, while `m_contract` is the actual `x` part in `x.member`.

Comment (axic) [reply libsolidity/ast/Types.cpp:1665]: Cool, didn't notice the type has `m_contract`.

**Comment by axic** [comment]
@chriseth updated, ready to merge

File: libsolidity/ast/Types.cpp (hunk 1 above)
Comment (axic) [reply libsolidity/ast/Types.cpp:1665]: Apparently `m_contract` doesn't behave correctly in this scenario for some reason.

Comment (chriseth) [reply libsolidity/ast/Types.cpp:1665]: Oh I'm sorry. Of course it has to be the source unit where it is used in, not where it was defined.

Comment (chriseth) [reply libsolidity/ast/Types.cpp:1665]: You initial comment was 100% correct.

**Comment by axic** [comment]
@chriseth passes tests now

**Review by chriseth — APPROVED** [review]

**Comment by chriseth** [comment]
I hope we don't hit the assert...

**Pull request merged**